services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/table/TableEntityManagerImpl.java

```diff
@@ -956,4 +944,26 @@ void linkVersionToTransaction(String tableIdString, long version, long transacti
 		return tableTransactionDao.getTransactionForVersion(tableId, version);
 	}
 
+
+	@WriteTransaction
+	@Override
+	public SnapshotResponse createTableSnapshot(UserInfo userInfo, String tableIdString, SnapshotRequest request) {
+		ValidateArgument.required(userInfo, "userInfo");
+		ValidateArgument.required(tableIdString, "TableId");
+		ValidateArgument.required(request, "request");
+		Long tableId = KeyFactory.stringToKey(tableIdString);
+		IdAndVersion idAndVersion = IdAndVersion.newBuilder().setId(tableId).build();
+		// Validate the user has permission to edit the table
+		tableManagerSupport.validateTableWriteAccess(userInfo, idAndVersion);
+		
+		Optional<Long> lastTransactionNumber = tableRowTruthDao.getLastTransactionId(tableIdString);
+		if(!lastTransactionNumber.isPresent()) {
+			throw new IllegalArgumentException("No transactions found for table: "+tableId);
```

Comment:
This table does not have a schema so a snapshot cannot be created. //comment about transactions